app/api/ada/hardwareWallet/newTransaction.js

```diff
@@ -154,3 +159,90 @@ function _outputAddressOrPath(out: TxOutput) {
   }
   return { address: out.address };
 }
+
+/** Generate a payload for Ledger SignTx */
+export async function createLedgerSignTxPayload(
+  txExt: UnsignedTransactionExt
+): Promise<LedgerSignTxPayload> {
+
+  // Transactions Hash
+  const txDataHexList = await txsBodiesForInputs(txExt.inputs);
+
+  // Inputs
+  const ledgerInputs: Array<LedgerInputTypeUTxO> =
+    _transformToLedgerInputs(txExt.inputs, txDataHexList);
+
+  // Outputs
+  const ledgerOutputs = _transformToLedgerOutputs(txExt.outputs);
+
+  return {
+    inputs: ledgerInputs,
+    outputs: ledgerOutputs,
+  };
+}
+
+function _transformToLedgerInputs(
+  inputs: Array<TxInput>, 
+  txDataHexList: Array<string>
+): Array<LedgerInputTypeUTxO> {
+  return inputs.map((input: TxInput, idx: number) => ({
+    txDataHex: txDataHexList[idx],
+    outputIndex: input.ptr.index,
+    path: _derivePath(input.addressing.change, input.addressing.index),
+  }));
+}
+
+function _transformToLedgerOutputs(
+  outputs: Array<TxOutput>
+): Array<LedgerOutputTypeAddress | LedgerOutputTypeChange> {
+  return outputs.map(x => ({
+    amountStr: x.value.toString(),
+    ..._ledgerOutputAddressOrPath(x)
+  }));
+}
+
+function _ledgerOutputAddressOrPath(out: TxOutput) {
+  if (out.isChange) {
+    // It's a change output
```

Comment:
I think we don’t need this comment because there is a line above that says isChange

Reply:
@nicarq sometimes i add comment to give some space to make it bit visibly clear and faster to understand.